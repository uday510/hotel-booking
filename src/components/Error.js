import React from 'react';

/**
 * Error component to display a "Page Not Found" message for invalid routes.
 * @component
 * @returns {JSX.Element} The JSX representation of the Error component.
 */
const Error = () => {
  /**
   * Renders the Error component with a custom error message.
   * @returns {JSX.Element} The JSX representation of the rendered component.
   */
  return (
    <div>
      <h2>Error: Page Not Found</h2>
      <p>The requested page does not exist.</p>
    </div>
  );
};

export default Error;
