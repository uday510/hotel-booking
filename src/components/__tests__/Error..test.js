/**
 * @file Error.test.js
 * @description Test file for the Error component using Jest and react-test-renderer.
 */

import renderer from 'react-test-renderer';
import React from 'react';
import Error from '../Error';

/**
 * Renders a tree using react-test-renderer.
 *
 * @param {ReactNode} tree - The React tree to be rendered.
 * @returns {ReactTestRenderer} The rendered tree.
 */
const renderTree = (tree) => renderer.create(tree);

describe('<Error>', () => {
  /**
   * Snapshot test for rendering the Error component.
   */
  it('should render component', () => {
    expect(renderTree(<Error />).toJSON()).toMatchSnapshot();
  });
});
