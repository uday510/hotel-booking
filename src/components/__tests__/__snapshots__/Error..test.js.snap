// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error> should render component 1`] = `
<div>
  <h2>
    Error: Page Not Found
  </h2>
  <p>
    The requested page does not exist.
  </p>
</div>
`;
